src/app/search/search-filters/stats.tsx

```diff
@@ -20,23 +32,32 @@ const statFilters: FilterDefinition[] = [
     // t('Filter.StatsExtras')
     description: tl('Filter.Stats'),
     format: 'stat',
-    suggestions: allStatNames,
-    filter: ({ filterValue }) => statFilterFromString(filterValue),
+    suggestions: allAtomicStats,
+    validateStat: (stat) =>
+      allAtomicStats.includes(stat) ||
+      stat.split(/&|\+/).every((s) => s !== 'any' && allAtomicStats.includes(s)),
+    filter: ({ filterValue, compare }) => statFilterFromString(filterValue, compare!),
```

Comment:
Because these filters are so powerful, they allow a bit more syntax than is convenient to statically declare, so I've opted for a dynamic validation callback on those, which is awkward but still allows validation without `FilterContext`.

These are probably also the most likely to be buggy wrt mismatches between declared and accepted syntax.